source/rpt/compartmentalization.cc

```diff
@@ -0,0 +1,963 @@
+#include <core/lethe_grid_tools.h>
+#include <core/solutions_output.h>
+
+#include <deal.II/dofs/dof_handler.h>
+
+#include <deal.II/grid/cell_id.h>
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_out.h>
+
+#include <deal.II/lac/block_vector.h>
+
+#include <deal.II/numerics/data_component_interpretation.h>
+#include <deal.II/numerics/data_out.h>
+
+#include <bits/stdc++.h>
+#include <rpt/compartmentalization.h>
+
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+template <int dim>
+Compartmentalization<dim>::Compartmentalization(
+  CPCalculatingParameters &CPparameters)
+  : computing_timer(std::cout, TimerOutput::summary, TimerOutput::wall_times)
+  , mpi_communicator(MPI_COMM_WORLD)
+  , triangulation(mpi_communicator)
+  , cp_parameters(CPparameters)
+{}
+
+template <int dim>
+void
+Compartmentalization<dim>::generate_cylindrical_grid()
+{
+  GridOut grid_out;
+  GridGenerator::subdivided_cylinder(
+    triangulation,
+    cp_parameters.cp_param.subdivisions,
+    cp_parameters.cp_param.cylinder_radius,
+    cp_parameters.cp_param.cylinder_half_length);
+  triangulation.refine_global(cp_parameters.cp_param.initial_refinement);
+  std::ofstream output("triangulation.vtk");
+  grid_out.write_vtk(triangulation, output);
+  std::vector<double>     cell_index_output;
+  std::vector<Point<dim>> points;
+  for (const auto &cell : triangulation.active_cell_iterators())
+    {
+      // std::vector<double> center;
+      Point<3, double> center;
```

Comment:
Is it only working is 3d? Here you don't use dim. If you don't plan to use it in 2D, you don't need to use the dim template :)